### PR TITLE
microcontroller: fix unsafe pattern usage

### LINKED ISSUE #619: trivial crash using microcontroller
program one with the following code:

```lua
sbi((A, B)
```

result:
```
AsyncErr: Lua: Runtime error from mod 'mesecons_microcontroller' in callback node_on_receive_fields(): ...minetest/mods/mesecons/mesecons_microcontroller/init.lua:422: unfinished capture
stack traceback:
	[C]: in function 'find'
	...minetest/mods/mesecons/mesecons_microcontroller/init.lua:422: in function 'command_sbi'
	...minetest/mods/mesecons/mesecons_microcontroller/init.lua:265: in function 'parsecode'
	...minetest/mods/mesecons/mesecons_microcontroller/init.lua:202: in function 'update'
	...minetest/mods/mesecons/mesecons_microcontroller/init.lua:141: in function <...minetest/mods/mesecons/mesecons_microcontroller/init.lua:106>
```

### PR DESCRIPTION
**Pull request by fluxionary**
particularly in a non-sandboxed environment.

fixes #619.

i'm far from certain these two cases are the only place this is done, but it's a start.

### CODE REVIEW

**Comment by sfan5** [comment]
Not sure what this has to do with executing code, suggested rename: "microcontroller: fix unsafe pattern usage"

File: mesecons_microcontroller/init.lua [hunk 1]
```diff
@@ -419,7 +420,8 @@ yc.command_sbi = function(params, eeprom, L, Lv)
 
 	if status == nil then return nil, nil end
 
-	if string.find("ABCD", params[1])~=nil and #params[1]==1 then --is a port
+	local b = params[1]:byte()
+	if #params[1]==1 and 65 <= b and b <= 68 then -- is a port
```
Comment (sfan5) [inline mesecons_microcontroller/init.lua:424]: I think you want the `#params[1] == 1` check before calling `byte` on it

Comment (fluxionary) [reply mesecons_microcontroller/init.lua:424]: i tested that it works whether the length is 0 or 1 or 2, but i admit it'd be clearer if i put the check first, will do.

**Comment by fluxionary** [comment]
> Not sure what this has to do with executing code, suggested rename: "microcontroller: fix unsafe pattern usage"

patterns are code, but i'll change it.

**Review by sfan5 — APPROVED** [review]
works

**Pull request merged**